stats/src/main/java/io/airlift/stats/JmxGcMonitor.java

```diff
@@ -140,6 +147,20 @@ private synchronized void onNotification(Notification notification)
                 long applicationRuntime = max(0, info.getStartTime() - lastGcEndTime);
                 lastGcEndTime = info.getEndTime();
 
+                if (info.getBeforeGcTotal().toBytes() / heapMemoryLimit > .8) {
+                    // Only warn of GC issues when the Heap is close to the limit
+                    double percentRecovered = 1 - info.getBeforeGcTotal().toBytes() / info.getAfterGcTotal().toBytes();
+                    if (percentRecovered < .1) {
+                        log.warn(
+                                "Major GC: application %sms, stopped %sms: %s -> %s (only recovered %s% of heap memory)",
```

Comment:
Probably want to round here. Also, need to use `%%` to render as `%`

`only recovered %.1f%% of heap memory`

Reply:
sounds good.